### ISSUE TITLE
./configure overwrites /dev/null

### ISSUE THREAD

**Issue by marc-chevalier** [ocamllibs/stdcompat#18]
When installing (with opam) in root and without sandbox, or manually (I have my reasons: I'm fighting with a terrible non-ocaml build system), the ./configure step overwrite `/dev/null`. To reproduce, as root:
```
export OPAMROOT=/opt/.opam  # For instance
opam init --bare --disable-sandboxing
opam switch create 4.14.0
opam install ocamlfind seq  # can also work with result or uchar instead of seq
# Everything is fine here
opam install stdcompat
# Now /dev/null is replaced by a normal file
```
I think I found the cause of that:
```
4028:      if $OCAMLFIND c -package result configure_test.ml -o /dev/null >/dev/null 2>&1; then :
4053:      if $OCAMLFIND c -package seq configure_test.ml -o /dev/null >/dev/null 2>&1; then :
4078:      if $OCAMLFIND c -package uchar configure_test.ml -o /dev/null >/dev/null 2>&1; then :
```
The `-o dev/null` seems dangerous in this case. Maybe one could remove it and change the next line from
```
rm configure_test.ml
```
to
```
rm configure_test.*
```
? Not sure this if this is good enough. But it seems to work for me. I can submit a PR if this fix suits you.

EDIT: I mean, change it in `configure.ac` of course.

**Comment by thierry-martinez** [comment]
Thank you very much, @marc-chevalier, for reporting this and sorry for the delay. I fixed this along the lines you described (but just removing the `-o` option and deleting all `configure_test.*` files are not enough because `ocamlc` outputs in `a.out` by default: instead, I replaced `-o /dev/null` by `-o configure_test.out`). I don't think this behavior of `ocamlc` is expected: I reported an OCaml issue.
